### ISSUE TITLE
Leaked IntentReceiver Exception using osmdroid

### ISSUE THREAD

**Issue by Fausto70** [osmdroid/osmdroid#343]
I have in my Android app a MapActivity, that shows map using osmdroid

The library I imported, inside my gradle.build file, is:
compile 'org.osmdroid:osmdroid-android:5.1@aar'

When I'm in this MapActivity and I go to another activity, I get always an exception, and I cannot understand how to avoid it, because I think that it is related to something done inside the library. As you can see in the following log, it happen after destroying of MapActivity

__________________________________
05-18 17:34:31.117  27073-27073/com.fpricoco.etip I/﹕ Classe 'MapActivity'  ----  'onDestroy()'  ----  Stato: Distrutta
05-18 17:34:31.157  27073-27073/com.fpricoco.etip E/ActivityThread﹕ Activity com.fpricoco.etip.Activities.MapActivity has leaked IntentReceiver org.osmdroid.tileprovider.modules.MapTileFileStorageProviderBase$MyBroadcastReceiver@43d85f60 that was originally registered here. Are you missing a call to unregisterReceiver()?
    android.app.IntentReceiverLeaked: Activity com.fpricoco.etip.Activities.MapActivity has leaked IntentReceiver org.osmdroid.tileprovider.modules.MapTileFileStorageProviderBase$MyBroadcastReceiver@43d85f60 that was originally registered here. Are you missing a call to unregisterReceiver()?
            at android.app.LoadedApk$ReceiverDispatcher.<init>(LoadedApk.java:792)
            at android.app.LoadedApk.getReceiverDispatcher(LoadedApk.java:593)
            at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1151)
            at android.app.ContextImpl.registerReceiver(ContextImpl.java:1138)
            at android.app.ContextImpl.registerReceiver(ContextImpl.java:1132)
            at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:365)
            at org.osmdroid.tileprovider.util.SimpleRegisterReceiver.registerReceiver(SimpleRegisterReceiver.java:21)
            at org.osmdroid.tileprovider.modules.MapTileFileStorageProviderBase.<init>(MapTileFileStorageProviderBase.java:34)
            at org.osmdroid.tileprovider.modules.MapTileFileArchiveProvider.<init>(MapTileFileArchiveProvider.java:57)
            at org.osmdroid.tileprovider.modules.MapTileFileArchiveProvider.<init>(MapTileFileArchiveProvider.java:76)
            at org.osmdroid.tileprovider.MapTileProviderBasic.<init>(MapTileProviderBasic.java:63)
            at org.osmdroid.tileprovider.MapTileProviderBasic.<init>(MapTileProviderBasic.java:41)
            at org.osmdroid.tileprovider.MapTileProviderBasic.<init>(MapTileProviderBasic.java:34)
            at org.osmdroid.views.overlay.MinimapOverlay.<init>(MinimapOverlay.java:114)
            at com.fpricoco.etip.Activities.MapActivity.onCreate(MapActivity.java:303)
            at android.app.Activity.performCreate(Activity.java:5008)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2034)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2095)
            at android.app.ActivityThread.access$600(ActivityThread.java:137)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1206)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:213)
            at android.app.ActivityThread.main(ActivityThread.java:4791)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:511)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:789)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:556)
            at dalvik.system.NativeStart.main(Native Method)
______________________________________

Line 303 of my code, where  I have the exception, : mMinimapOverlay = new MinimapOverlay(myContext, myMapView.getTileRequestCompleteHandler()); 
______________________________________

Searching on Google I found that I have to "unregisterReceiver(yourReceiver)", the point is that I don't have in my case any "yourReceiver" setted in my code, I think it is done inside library.
UnregisterReceiver() wants as argument an object of type BroadcastReceiver; the point is that I don't have any object of this type in my code. 

I also found this solution "https://groups.google.com/forum/#!topic/osmdroid/wDEvFpWfTIU" , but I don't understand how to detach.

______________________________________
Finally I also tried in this way, but it didn't work:

// .. get custom tiles off the screen
myMapView.getOverlays().remove(mMinimapOverlay);
// .. release all custom tiles' bitmaps
myMapView.getTileProvider().clearTileCache();
// .. unregister intent receiver
myMapView.getTileProvider().detach();
______________________________________

.

**Comment by spyhunter99** [comment]
Did you add the MapView to your view programmatically or using an xml layout?

**Comment by spyhunter99** [comment]
If you're programmatically creating the map view, you'll have to handle detach events yourself.

myMapView.onDetach();

**Comment by Fausto70** [comment]
Hi,
first of all thank you.

I added programmatically, infact I have in my xml:
           
           <org.osmdroid.views.MapView
            android:id="@+id/id_Map__Mapview"
            android:layout_width="match_parent"
            android:layout_height="match_parent"
            android:longClickable="true"
             />

Then inside the Activity, in onCreate() I have:
```
            myMapView = (MapView) findViewById(R.id.id_Map__Mapview);
            mMinimapOverlay = new MinimapOverlay(myContext, myMapView.getTileRequestCompleteHandler());

```
But I didn't understand how to apply your suggestion; I tried putting "myMapView.onDetach();
" inside onPause(), but I continue to get the error.
Can you please clarify better? I'm sorry but I'm not very expert...

ps: I'm searching a consultant for the OSM part of my app, I have some issues to fix, you are very expert, may you write me to fpricoco@yahoo.it?

Thank you in advance
Fausto

**Comment by Fausto70** [comment]
@spyhunter99   
I added your nick hoping you will be notified in this way...

**Comment by spyhunter99** [comment]
If the map view is defined in xml, then you shouldn't be having any of these problems. If your variable 'myMapView' a static variable? Is your map view within a fragment?

**Comment by Fausto70** [comment]
@spyhunter99 
Hi again,
thanks for your reply. 
1) myMapView is static, it is a problem?
2) my map is not within a fragment; should it be?

by the way, any way to get in contact with you an get your (payed) consultancy? my email is in previous message...
Fausto

**Comment by spyhunter99** [comment]
You should never have anything with context or that inherits from View be
static. That's probably the issue

**Comment by Fausto70** [comment]
@spyhunter99 
Hi, I tried putting myMapView not static, but nothing changed, I continue to get the error. 
Any chance to send you by email my  .java file and get your help? With your expertize I think it should be easy for you...

Thank you in advance
Fausto

**Comment by spyhunter99** [comment]
Go ahead

**Comment by Fausto70** [comment]
@spyhunter99 
Hi again, I just replied to your mail; sorry for delay in reply, but I spent some time to write the email to be as precise as possible in describing the issues

**Comment by spyhunter99** [comment]
@Fausto70 everything has been merged into master. Please try to replicate the error condition. I'm unable to nor do i know the exact conditions in which it occurs

**Comment by spyhunter99** [comment]
operation time out. reopen if this is still an issue for you
